ekncontent/configure.ac

```diff
@@ -0,0 +1,196 @@
+AC_PATH_PROG([GIO_QUERYMODULES], [gio-querymodules], [notfound])
+AS_IF([test x$GIO_QUERYMODULES = xnotfound], [AC_MSG_ERROR([gio-querymodules not found.
+Try installing "glib" package.])])
+
+AC_CACHE_SAVE
+
+# Check that the compiler really supports C99
+AS_IF([test "x$ac_cv_prog_cc_c99" = "xno"], [AC_MSG_ERROR([C99 is required.])])
+
+# Configure options
+# -----------------
+# --enable-strict-flags: Compile with strict compiler flags. Done automatically
+# during 'make distcheck'.
+AC_ARG_ENABLE([strict-flags],
+    [AS_HELP_STRING([--enable-strict-flags=@<:@no/yes/error@:>@],
+        [Use strict compiler flags @<:@default=no@:>@])],
+    [],
+    [enable_strict_flags=no])
+# Emmanuele's list of flags
+STRICT_COMPILER_FLAGS="$STRICT_COMPILER_FLAGS
+    -Wall
+    -Wcast-align
+    -Wuninitialized
+    -Wno-strict-aliasing
+    -Werror=pointer-arith
+    -Werror=missing-declarations
+    -Werror=redundant-decls
+    -Werror=empty-body
+    -Werror=format
+    -Werror=format-security
+    -Werror=format-nonliteral
+    -Werror=init-self"
+AS_CASE([$enable_strict_flags],
+    [yes],
+        [AS_COMPILER_FLAGS([STRICT_CFLAGS], [$STRICT_COMPILER_FLAGS])],
+    [no],
+        [],
+        [error],
+        [
+            STRICT_COMPILER_FLAGS="$STRICT_COMPILER_FLAGS -Werror"
+            AS_COMPILER_FLAGS([STRICT_CFLAGS], [$STRICT_COMPILER_FLAGS])
+        ],
+    [AC_MSG_ERROR([Invalid option for --enable-strict-flags])])
+AC_SUBST([STRICT_CFLAGS])
+
+# For 'make dist' or 'make distcheck', --enable-gtk-doc is required
+AC_MSG_CHECKING([whether this configuration allows building distributions])
+can_make_dist=yes
+AM_COND_IF([ENABLE_GTK_DOC], [], [can_make_dist=no])
+AM_CONDITIONAL([CAN_MAKE_DIST], [test "x$can_make_dist" = "xyes"])
+AC_MSG_RESULT([$can_make_dist])
+
+# JASMINE_JUNIT_REPORTS_DIR: Where to put test reports
+AC_MSG_CHECKING([where to put test reports])
+AC_ARG_VAR([JASMINE_JUNIT_REPORTS_DIR], [Where to put test reports])
+AS_IF([test -n "$JASMINE_JUNIT_REPORTS_DIR"],
+    [JASMINE_REPORT_ARGUMENT="--junit $JASMINE_JUNIT_REPORTS_DIR/\$\${log/%.log/.js.xml}"
+    AC_MSG_RESULT([in $JASMINE_JUNIT_REPORTS_DIR])],
+    [JASMINE_REPORT_ARGUMENT=
+    AC_MSG_RESULT([nowhere])])
+AC_SUBST([JASMINE_REPORT_ARGUMENT])
+
+# ------------------
+# These go into the pkg-config file as Requires: and Requires.private:
+# (Generally, use Requires.private: instead of Requires:)
+# Update these whenever you use a function that requires a certain API version
+EKN_CONTENT_REQUIRED_MODULES=
+EKN_CONTENT_REQUIRED_MODULES_PRIVATE="eos-shard-0 >= 0.1.2 gio-2.0 glib-2.0 gobject-2.0"
+AC_SUBST(EKN_CONTENT_REQUIRED_MODULES)
+AC_SUBST(EKN_CONTENT_REQUIRED_MODULES_PRIVATE)
+PKG_CHECK_MODULES([EKN_CONTENT], [
+    $EKN_CONTENT_REQUIRED_MODULES
+    $EKN_CONTENT_REQUIRED_MODULES_PRIVATE
+])
+
+PKG_CHECK_MODULES([EKN_VFS], [
+    eos-shard-0
+    gio-2.0
+    gmodule-2.0
+])
+
+# Code Coverage
+# -------------
+EOS_COVERAGE_REPORT([c])
```

Comment:
I would go ahead and switch to [`AX_CODE_COVERAGE`](https://www.gnu.org/software/autoconf-archive/ax_code_coverage.html) entirely here